packages/lit-html/src/test/directives/ref_test.ts

```diff
@@ -154,4 +149,47 @@ suite('ref', () => {
     assert.equal(divRef1.value, div);
     assert.equal(divRef2.value, div);
   });
+
+  test('two ref callbacks alternating', () => {
```

Comment:
Add a test to verify the refs are always set in tree order:

```
html`<div id="first" ${ref(myRef)}></div><div id="next" ${ref(myRef)}>${html`<span id="last" ${ref(myRef)}`}</div>`

Reply:
Good call. This is a weird usage of ref because it's last-in-wins (aka there's no point in 'first' and 'next' refs since they'll always be overwritten), but we weren't calling callbacks correctly in this setup (if `myRef` was a callback, it should call all 3 in order each render I think...?). Added tests for that.